### PR TITLE
Bug 1821427: Giving Image Config resource its own controller

### PR DESCRIPTION
**Pull request by ricardomaraschini**
Created a separated controller to manager image config resource.

### CODE REVIEW

File: pkg/operator/imageconfig.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+func NewImageConfigController(
+	configClient configset.ConfigV1Interface,
+	routeInformer routev1informers.RouteInformer,
+	serviceInformer corev1informers.ServiceInformer,
+) *ImageConfigController {
+	icc := &ImageConfigController{
+		configClient: configClient,
+		routeLister: routeInformer.Lister().Routes(
+			defaults.ImageRegistryOperatorNamespace,
+		),
```
Comment (dmage) [inline pkg/operator/imageconfig.go:52]: Are you trying to fit into 80 columns?

Comment (ricardomaraschini) [reply pkg/operator/imageconfig.go:52]: Sorta. In the end just put a comment where you feel like I should not indent and I will fix it for you.

File: pkg/operator/imageconfig.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+// sync keeps image.config.openshift.io/cluster status updated.
+func (icc *ImageConfigController) sync() error {
+	cfg, err := icc.configClient.Images().Get(
+		context.TODO(), defaults.ImageConfigName, metav1.GetOptions{},
+	)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	// This resource must always exist.
```
Comment (dmage) [inline pkg/operator/imageconfig.go:138]: What does this comment mean?

Comment (ricardomaraschini) [reply pkg/operator/imageconfig.go]: Mental note I forgot to remove. Removed now.

**Pull request merged**